### ISSUE TITLE
Panic at 'sfnt: table not found'

### ISSUE THREAD

**Issue by mulimoen** [crlf0710/tectonic#109]
Running tectonic on a s simple hello world tex document, and does not provide a pdf document.

**Comment by mulimoen** [comment]
Running git bisect

**Comment by mulimoen** [comment]
bad: bcff88969f6c50adeffcc341970def0cc0499cd8
good: 84a99aa1ca3caf9edca66f8b8163e268c9d331be

**Comment by mulimoen** [comment]
@burrbull want to have a look?

**Comment by mulimoen** [comment]
There is an off-by-one error in dpx_cidtype0.rs, with
`for j in (0..7).rev()` which should be `for j in (0..=7).rev()`

**Comment by cormacrelf** [comment]
Aw man, just beat me to it on that one

**Comment by burrbull** [comment]
> There is an off-by-one error in dpx_cidtype0.rs, with
> `for j in (0..7).rev()` which should be `for j in (0..=7).rev()`

How you did it? I'm amazed.

**Comment by cormacrelf** [comment]
I didn't spot anything else, hopefully that's it.

**Comment by mulimoen** [comment]
There is still a panic, but that could be in another commit

**Comment by crlf0710** [comment]
Mmm, is there some way that can cover this with tests?

**Comment by mulimoen** [comment]
We need a full integration test where we run the resulting binary

**Comment by crlf0710** [comment]
I'm debugging, and it seems this code (in dpx_sfnt.rs) is wrong:
```
unsafe extern "C" fn convert_tag(tag: &mut [u8; 4], mut u_tag: u32) {
    for i in 3..=0 {
```
this loop won't run.

**Comment by mulimoen** [comment]
Ranges always go forwards. There should be a rev() on it `(0..=3).rev()`

**Comment by burrbull** [comment]
https://github.com/crlf0710/tectonic/commit/b712e0eab3873601b784d0fbcc9492ba5531f21a

**Comment by crlf0710** [comment]
just call `to_be_bytes` will be fine. PRs welcome! Besides, is there a clippy lint for this?

**Comment by mulimoen** [comment]
There is the `reverse_range_loop` lint, and clippy errors out on the codebase.

**Comment by mulimoen** [comment]
Fixed with #110 #112
